components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/AccessTokenIssuer.java

```diff
@@ -1097,4 +1103,35 @@ private static boolean isNotActiveState(String appState) {
 
         return !APP_STATE_ACTIVE.equalsIgnoreCase(appState);
     }
+
+    /**
+     * If the user is not found in the given tenant domain, check the user existence from ancestor organizations and
+     * provide the correct user store manager.
+     *
+     * @param tenantDomain The tenant domain of the authenticated user.
+     * @param userId The ID of the authenticated user.
+     * @return User store manager of the user reside organization.
+     */
+    private Optional<AbstractUserStoreManager> getUserStoreManagerFromUserResideOrganization(String tenantDomain,
```

Comment:
What if the user is in a secondary userstore? This method will be loading the primary userstore manager right?
Is that other user store managers will be recursively searched?

Reply:
The primary user store manager has the ability to check the user existence in secondary user store chain.
The following set up was tested.
1 - Mark username as the subject identifier in the console.
2 - Plug a secondary user store and assign admin permission for a user.
3 -  Login to the console with that user, and create a suborganization.
4 - Switch to that suborganization.

The ID token was as follows. The username has been resolved properly and could switch to the suborganization successfully.

<img width="554" alt="Screenshot 2023-06-30 at 07 20 55" src="https://github.com/wso2-extensions/identity-inbound-auth-oauth/assets/35717390/52135842-719c-4b7b-a91e-408f52f6e200">

Reply:
Fixed by - https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2103/commits/1a9a3b2544c31dde5838f01e845435c8937503c8